node/src/accountant/db_access_objects/failed_payable_dao.rs

```diff
@@ -0,0 +1,778 @@
+                        "Only {} out of {} records inserted",
+                        inserted_rows,
+                        txs.len()
+                    )))
+                }
+            }
+            Err(e) => Err(FailedPayableDaoError::SqlExecutionFailed(e.to_string())),
+        }
+    }
+
+    fn retrieve_txs(&self, condition: Option<FailureRetrieveCondition>) -> Vec<FailedTx> {
+        let raw_sql = "SELECT tx_hash, receiver_address, amount_high_b, amount_low_b, \
+        timestamp, gas_price_wei_high_b, gas_price_wei_low_b, nonce, reason, checked FROM failed_payable"
+            .to_string();
+        let sql = match condition {
+            None => raw_sql,
+            Some(condition) => format!("{} {}", raw_sql, condition),
+        };
+
+        let mut stmt = self
+            .conn
+            .prepare(&sql)
+            .expect("Failed to prepare SQL statement");
+
+        stmt.query_map([], |row| {
+            let tx_hash_str: String = row.get(0).expectv("tx_hash");
+            let hash = TxHash::from_str(&tx_hash_str[2..]).expect("Failed to parse TxHash");
+            let receiver_address_str: String = row.get(1).expectv("receiver_address");
+            let receiver_address =
+                Address::from_str(&receiver_address_str[2..]).expect("Failed to parse Address");
+            let amount_high_b = row.get(2).expectv("amount_high_b");
+            let amount_low_b = row.get(3).expectv("amount_low_b");
+            let amount = BigIntDivider::reconstitute(amount_high_b, amount_low_b) as u128;
+            let timestamp = row.get(4).expectv("timestamp");
+            let gas_price_wei_high_b = row.get(5).expectv("gas_price_wei_high_b");
+            let gas_price_wei_low_b = row.get(6).expectv("gas_price_wei_low_b");
+            let gas_price_wei =
+                BigIntDivider::reconstitute(gas_price_wei_high_b, gas_price_wei_low_b) as u128;
+            let nonce = row.get(7).expectv("nonce");
+            let reason_str: String = row.get(8).expectv("reason");
+            let reason =
+                FailureReason::from_str(&reason_str).expect("Failed to parse FailureReason");
+            let checked: u8 = row.get(9).expectv("checked");
+
+            Ok(FailedTx {
+                hash,
+                receiver_address,
+                amount,
+                timestamp,
+                gas_price_wei,
+                nonce,
+                reason,
+                checked,
+            })
+        })
+        .expect("Failed to execute query")
+        .filter_map(Result::ok)
+        .collect()
+    }
+
+    fn update_tx_failures(&self, hash_map: &FailureUpdates) -> Result<(), FailedPayableDaoError> {
+        if hash_map.is_empty() {
+            return Err(FailedPayableDaoError::EmptyInput);
+        }
+
+        for (hash, failure_reason) in hash_map {
+            let sql = format!(
+                "UPDATE failed_payable SET reason = '{:?}' WHERE tx_hash = '{:?}'",
```

Comment:
This is not what I expected.

First, why would we create a record with a wrong failure reason? More over, why would we get into a situation where we want to correct our previous opinion - the former reason?

Second, my understanding is that the only update that would ever happen on this table will be checking off the already back-checked transactions. So from 0 to 1, I suppose. I really cannot think of a different reason which brings me to saying that maybe we don't need such a generically looking method, but it could simply be `check_off_pending_too_long`.

However, go ahead and leave this generically considered interface. Only change it to a different modification in the table. Yours isn't useful in our use case. I hope to be right...